cache/ttlmemory.go

```diff
@@ -0,0 +1,57 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+// NewTTLInMemory creates a new thread-safe in-memory ttl cache.
+// This is most useful when storing tokens that will be handed off to
+// other services.
+func NewTTLInMemory[T any](name string) TTLCache[T] {
+	return newTTLOtelCache(&ttlInMemoryCache[T]{
+		_name: name,
+		mu:    sync.RWMutex{},
+		cache: make(map[string]ttlValue[T]),
+	})
+}
+
+type ttlInMemoryCache[T any] struct {
+	_name string
```

Comment:
nit: can we just call this `name`?

Reply:
I removed this but it wasn't because the method was called `name` and you can't have a struct value and a method with the same name